feat(broker): featured broker events logic

Voronapxl opened the pull request:
1. Featured logic for events when scenarios are updated with objects or zones, F22 cache will be terminated for such events

Voronapxl committed:
feat(broker):
1. Continuation of broker logic features

Voronapxl committed:
Merge branch 'dev' of https://github.com/IDUclub/EffectsAPI into feat/broker

Voronapxl committed:
feat(broker):
1.Featured logic for events when scenarios are updated with objects or zones, F22 cache will be terminated for such events

kanootoko commented on app/broker_handlers/scenario_updated_handler.py:
```diff
@@ -0,0 +1,133 @@
+                f"Cache invalidation applied: method={rule.method} owner_id={owner_id} deleted_files={deleted}"
+            )
+
+        return total_deleted
```
Это ведь просто то же самое, что и в `cache_invalidation`?

kanootoko commented on app/broker_handlers/scenario_updated_handler.py:
```diff
@@ -0,0 +1,133 @@
+        return total_deleted
+
+
+class CacheInvalidationHandlerCore:
```
Тут не должно быть инкапсуляции/наследования в какую-нибудь сторону с `BaseCacheInvalidationHandler`? А то код-то одинаковый, кажется

kanootoko commented on app/broker_handlers/scenario_updated_handler.py:
```diff
@@ -0,0 +1,133 @@
+        pass
+
+    async def handle(self, event: EventT, ctx: Message = None):
+        return await self._core.process(event)
```
`ScenarioObjectsUpdatedHandler` и `ScenarioZonesUpdatedHandler` ведь одинаковые полностью по коду?
Может, можно сразу от двух классов отнаследоваться, или реализовать логику в миксине, а эти два класса отнаследовать от правильных `BaseMessageHandler`'ов и миксина?

kanootoko commented on app/effects_api/modules/task_service.py:
```diff
@@ -240,10 +240,11 @@ async def stop(self):
 async def lifespan(app: FastAPI):
     worker.start()
+    await consumer.start(["scenario.events"])
+    await producer.start()
```
После yield'а у воркера есть `stop()`. А почему этих не нужно останавливать?

Voronapxl committed:
fix(broker):
1. Deletion of redundant broker logic for cache invalidation
2. Featured extermination of consumer and producer workers

soldey commented on app/broker_handlers/cache_invalidation.py:
```diff
@@ -0,0 +1,74 @@
+            f"scenario_id={getattr(event, 'scenario_id', None)}"
+        )
+
+        total_deleted = self._invalidation_service.invalidate(event, self._rules)
```
сделать await asyncio.to_thread

Voronapxl committed:
fix(broker):
1. invalidation methods putted to thread

kanootoko commented on app/broker_handlers/scenario_updated_handler.py:
```diff
@@ -104,24 +30,13 @@ async def on_shutdown(self):
     async def handle(self, event: EventT, ctx: Message = None):
-        return await self._core.process(event)
-
+        return await self._handle_cache_invalidation(event, ctx)
```
Мне казалось, суть Mixin'а в том, чтобы придать классу поведение (то есть сразу в нем создать метод `handle`, а в наследниках вообще ничего не трогать; менее прозрачно, но если вызывать методы самостоятельно, то можно было и просто инкапсуляцией ограничиться.

Не критично, но странновато

Voronapxl committed:
feat(producer):
1. KafkaProducerClient init_loop set to False

Voronapxl committed:
feat(producer):
1. KafkaProducerClient init_loop at start
